test/e2e_node/runtime_conformance_test.go

```diff
@@ -28,111 +32,319 @@ import (
 )
 
 const (
-	retryTimeout = time.Minute * 4
+	retryTimeout = time.Minute * 5
 	pollInterval = time.Second * 5
 )
 
-var _ = Describe("Container Runtime Conformance Test", func() {
+type testStatus struct {
+	Name             string
+	RestartPolicy    api.RestartPolicy
+	Phase            api.PodPhase
+	State            ContainerState
+	RestartCountOper string
+	RestartCount     int32
+	Ready            bool
+}
+
+var _ = Describe("Container runtime Conformance Test", func() {
 	var cl *client.Client
 
 	BeforeEach(func() {
 		// Setup the apiserver client
 		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
 	})
 
-	Describe("container conformance blackbox test", func() {
-		Context("when running a container that terminates", func() {
-			var containerCase ConformanceContainer
+	Describe("container runtime conformance blackbox test", func() {
+		namespace := "runtime-conformance"
+		Context("when start a container that exits successfully", func() {
+			var testCContainers []ConformanceContainer
 
 			BeforeEach(func() {
-				containerCase = ConformanceContainer{
-					Container: api.Container{
-						Image:           "gcr.io/google_containers/busybox",
-						Name:            "busybox",
-						Command:         []string{"sh", "-c", "env"},
-						ImagePullPolicy: api.PullIfNotPresent,
+				testCContainers = []ConformanceContainer{}
+			})
+
+			It("it should run with the expected status [Conformance]", func() {
+				testContainer := api.Container{
+					Image: "gcr.io/google_containers/busybox",
+					VolumeMounts: []api.VolumeMount{
+						{
+							MountPath: "/restart-count",
+							Name:      "restart-count",
+						},
 					},
-					Client:   cl,
-					Phase:    api.PodSucceeded,
-					NodeName: *nodeName,
+					ImagePullPolicy: api.PullIfNotPresent,
+				}
+				testVolumes := []api.Volume{
+					{
+						Name: "restart-count",
+						VolumeSource: api.VolumeSource{
+							HostPath: &api.HostPathVolumeSource{
+								Path: os.TempDir(),
+							},
+						},
+					},
+				}
+				testCount := int32(3)
+				testStatuss := []testStatus{
+					{"terminate-always", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, ">", testCount, false},
```

Comment:
Oh no, I remembered why I do this. In the 'restart always' sub case, the pod phase will change between 'waiting-running-succeed''. That is why I add codes like:
```
ContainerStateWaiting    ContainerState = 1
ContainerStateRunning    ContainerState = 1 << 1
```

Since we are testing container behaviors , so we just need to confirm if container state is match. The expectation of PodPhase in runtime_conformance_test.go is just used to reach the test condition (like skip the pending phase). So I also prefer to use the current implementation.